### ISSUE TITLE
VSTHRD003UseJtfRunAsyncAnalyzer: Analyzer failure while processing syntax

### ISSUE THREAD

**Issue by davkean** [microsoft/vs-threading#573]
#### Bug description
VSTHRD003UseJtfRunAsyncAnalyzer crashes

#### Repro steps
``` C#
        /// <summary>
        /// Gets the value of the <see cref="AssemblyDescriptionAttribute"/> that is embedded in the referenced assembly.
        /// </summary>
        public Task<string> GetDescriptionAsync()
        {
            // UNDONE (andarno): return the value of this attribute from the target assembly:
            // System.Reflection.AssemblyDescriptionAttribute
            // Hopefully RAR can do this for us.
            return TaskResults.EmptyString;
        }

   internal static class TaskResults
    {
        /// <summary>
        /// A completed task to return empty string.
        /// </summary>
        internal static readonly Task<string> EmptyString = Task.FromResult(string.Empty);
    }
```

#### Expected behavior
No crash

#### Actual behavior
```
Severity	Code	Description	Project	File	Line	Suppression State
Warning	AD0001	Analyzer 'Microsoft.VisualStudio.Threading.Analyzers.VSTHRD003UseJtfRunAsyncAnalyzer' threw an exception of type 'System.Exception' with message 'Analyzer failure while processing syntax at C:\CPS\src\Microsoft.VisualStudio.ProjectSystem.Implementation\References\UnresolvedReferenceBase.cs(206,13): System.ArgumentException SyntaxTree is not part of the compilation
Parameter name: syntaxTree. Syntax: return TaskResults.EmptyString;'.	Microsoft.VisualStudio.ProjectSystem.Implementation		1	Active
Warning	AD0001	Analyzer 'Microsoft.VisualStudio.Threading.Analyzers.VSTHRD003UseJtfRunAsyncAnalyzer' threw an exception of type 'System.Exception' with message 'Analyzer failure while processing syntax at C:\CPS\src\Microsoft.VisualStudio.ProjectSystem.Implementation\References\ExpandedSdkReference.cs(75,13): System.ArgumentException SyntaxTree is not part of the compilation
Parameter name: syntaxTree. Syntax: return TaskResults.EmptyString;'.	Microsoft.VisualStudio.ProjectSystem.Implementation		1	Active
Warning	AD0001	Analyzer 'Microsoft.VisualStudio.Threading.Analyzers.VSTHRD003UseJtfRunAsyncAnalyzer' threw an exception of type 'System.Exception' with message 'Analyzer failure while processing syntax at C:\CPS\src\Microsoft.VisualStudio.ProjectSystem.Implementation\Properties\ProjectInstanceItemTypeProperties.cs(71,17): System.ArgumentException SyntaxTree is not part of the compilation
Parameter name: syntaxTree. Syntax: return TaskResults.EmptyString;'.	Microsoft.VisualStudio.ProjectSystem.Implementation		1	Active

```

- Version used: Microsoft.VisualStudio.Threading.Analyzers 16.4.36

**Comment by AArnott** [comment]
Ouch. This looks like my fix for #472 was completely ineffective.

**Comment by davkean** [comment]
What version was this previously fixed in? I noticed that CPS was using 16.3 and hadn't seen this before.

**Comment by AArnott** [comment]
It was fixed in the PR that completed just before the perf fix. So 16.4.34 was the build that contained the fix for this (or so I thought). I'm reviewing the tests now to see if they are somehow ineffective.

**Comment by AArnott** [comment]
I can repro the problem in a real solution (though the test passes) so obviously the test isn't testing what I thought.

**Comment by AArnott** [comment]
My repro vanished. I'm not sure how yet.

**Comment by davkean** [comment]
@333fred and @mavasani were looking at trying to repro this.

**Comment by AArnott** [comment]
False alarm. David wasn't testing with 16.4.36. He was testing with a 16.5 prerelease that doesn't have the fix to #472.
